src/coreclr/jit/morph.cpp

```diff
@@ -12991,9 +12991,12 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 GenTree* addr = commaNode->AsOp()->gtOp2;
                 // TODO-1stClassStructs: we often create a struct IND without a handle, fix it.
                 op1 = gtNewIndir(typ, addr);
-                // This is very conservative
-                op1->gtFlags |= treeFlags & ~GTF_ALL_EFFECT & ~GTF_IND_NONFAULTING;
+
+                // Determine flags on the indir
+                //
+                op1->gtFlags |= treeFlags & ~GTF_ALL_EFFECT;
                 op1->gtFlags |= (addr->gtFlags & GTF_ALL_EFFECT);
+                op1->gtFlags |= treeFlags & (GTF_GLOB_REF | GTF_IND_FLAGS);
```

Comment:
Is the `GTF_IND_FLAGS` portion needed? Seems like it'd be set by the `|= treeFlags & ~GTF_ALL_EFFECT;` above.

Reply:
> Is the `GTF_IND_FLAGS` portion needed?

No, it's not needed.